modules/core/src/utils/filesystem.cpp

```diff
@@ -88,10 +88,13 @@ cv::String join(const cv::String& base, const cv::String& path)
 cv::String canonical(const cv::String& path)
 {
     cv::String result;
-#ifdef _WIN32
-    const char* result_str = _fullpath(NULL, path.c_str(), 0);
+    const char* result_str =
+#ifdef NO_GETENV
+        NULL;
+#elif defined(_WIN32)
+        _fullpath(NULL, path.c_str(), 0);
 #else
-    const char* result_str = realpath(path.c_str(), NULL);
+        realpath(path.c_str(), NULL);
```

Comment:
This code is guarded by `OPENCV_HAVE_FILESYSTEM_SUPPORT`.
Perhaps it make sense to disable this support instead (see .hpp file).

> #ifdef NO_GETENV
> ...
> _fullpath

There is inconsistency with semantic.

Reply:
Hm, Windows CE operates on absolute paths only. So converting a path to a canonical absolute path in this case is a noop.  That doesn't mean that there is no filesystem support. In this case, the path to be converted should be returned as is, just as the function does on L104. 

There is an inconsistency with the `OPENCV_HAVE_FILESYSTEM_SUPPORT`. It seems to disable filesystem functionality for WINRT by default, but still have #ifdefs in the code paths. Perhaps `OPENCV_HAVE_FILESYSTEM_SUPPORT` means that all functions have not been fully ported yet? In that case, I can revert this change and perhaps send in new PR's that address that separately.